bt/tasks/utility/bt_evaluate_expression.cpp

```diff
@@ -0,0 +1,168 @@
+	Object *obj = node_param->get_value(get_agent(), get_blackboard());
+	ERR_FAIL_COND_V_MSG(obj == nullptr, FAILURE, "BTEvaluateExpression: Failed to get object: " + node_param->to_string());
+	ERR_FAIL_COND_V_MSG(is_parsed != Error::OK, FAILURE, "BTEvaluateExpression: Failed to parse expression: " + expression.get_error_text());
+
+	if (input_include_delta) {
+		processed_input_values[0] = p_delta;
+	}
+	for (int i = int(input_include_delta); i < input_values.size(); ++i) {
+		const Ref<BBVariant> &bb_variant = input_values[i];
+		processed_input_values[i] = bb_variant->get_value(get_agent(), get_blackboard());
+	}
+
+	Variant result = expression.execute(processed_input_values, obj, false);
+	ERR_FAIL_COND_V_MSG(expression.has_execute_failed(), FAILURE, "BTEvaluateExpression: Failed to execute: " + expression.get_error_text());
+
+	if (!result_var.is_empty()) {
+		get_blackboard()->set_var(result_var, result);
+	}
+
+	return SUCCESS;
+}
+
+//**** Godot
+
+void BTEvaluateExpression::_bind_methods() {
+	ClassDB::bind_method(D_METHOD("parse"), &BTEvaluateExpression::parse);
+	ClassDB::bind_method(D_METHOD("set_expression_string", "p_method"), &BTEvaluateExpression::set_expression_string);
+	ClassDB::bind_method(D_METHOD("get_expression_string"), &BTEvaluateExpression::get_expression_string);
+	ClassDB::bind_method(D_METHOD("set_node_param", "p_param"), &BTEvaluateExpression::set_node_param);
+	ClassDB::bind_method(D_METHOD("get_node_param"), &BTEvaluateExpression::get_node_param);
+	ClassDB::bind_method(D_METHOD("set_input_vars", "p_input_vars"), &BTEvaluateExpression::set_input_vars);
+	ClassDB::bind_method(D_METHOD("get_input_vars"), &BTEvaluateExpression::get_input_vars);
+	ClassDB::bind_method(D_METHOD("set_input_values", "p_input_values"), &BTEvaluateExpression::set_input_values);
+	ClassDB::bind_method(D_METHOD("get_input_values"), &BTEvaluateExpression::get_input_values);
+	ClassDB::bind_method(D_METHOD("set_input_include_delta", "p_input_include_delta"), &BTEvaluateExpression::set_input_include_delta);
+	ClassDB::bind_method(D_METHOD("is_input_delta_included"), &BTEvaluateExpression::is_input_delta_included);
+	ClassDB::bind_method(D_METHOD("set_result_var", "p_result_var"), &BTEvaluateExpression::set_result_var);
+	ClassDB::bind_method(D_METHOD("get_result_var"), &BTEvaluateExpression::get_result_var);
+
+	ADD_PROPERTY(PropertyInfo(Variant::OBJECT, "node", PROPERTY_HINT_RESOURCE_TYPE, "BBNode"), "set_node_param", "get_node_param");
+	ADD_PROPERTY(PropertyInfo(Variant::STRING, "expression_string"), "set_expression_string", "get_expression_string");
+	ADD_PROPERTY(PropertyInfo(Variant::STRING, "result_var"), "set_result_var", "get_result_var");
+	ADD_GROUP("Inputs", "input_");
+	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "input_include_delta"), "set_input_include_delta", "is_input_delta_included");
+	ADD_PROPERTY(PropertyInfo(Variant::PACKED_STRING_ARRAY, "input_vars", PROPERTY_HINT_ARRAY_TYPE, "String"), "set_input_vars", "get_input_vars");
```

Comment:
Calling this property `input_vars` is problematic, because we use `var` to signify blackboard variables. It becomes confusing what an array of strings called like that means. Is it a list of blackboard variables that serve as the input here, or a list of input names for the Expression?

I looked at `Expression` class, and parameter is called `input_names` there. Let's call it like that. It'll be **Names** and **Values** in the inspector and no overlapping terminology with the blackboard variables.